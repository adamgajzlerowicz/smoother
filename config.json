{
    "commands": [
        {
            "name": "update react",
            "command": "gitCommand",
            "fields": [],
            "payload": {
                "path": "/Users/adam/projects/react",
                "branch": "master",
                "pull": true
            }
        },
        {
            "name": "run mock command",
            "command": "anyCommand",
            "payload": {
                "command": "node /Users/adam/projects/smoother/packages/mockEmiter/index.js"
            }
        },
        {
            "name": "this will fail",
            "command": "anyCommand",
            "payload": {
                "command": "node /projects/smoother/packages/mockEmiter/index.js"
            }
        },
        {
            "name": "b -> a",
            "command": "replaceCommand",
            "payload": {
                "path": "node /projects/smoother/packages/mockEmiter/index.js",
                "replacer": "a",
                "replacee": "b"
            }
        },
        {
            "name": "a -> b",
            "command": "replaceCommand",
            "payload": {
                "path": "node /projects/smoother/packages/mockEmiter/index.js",
                "replacer": "a",
                "replacee": "b"
            }
        }
    ]
}
