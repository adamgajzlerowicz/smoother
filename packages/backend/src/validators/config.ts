import * as Joi from 'Joi';

import { schema } from './schema';

const options = {
    stripUnknown: true
};

const validateConfig = config => Joi.validate(config, schema, options);

export {
    validateConfig,
};

