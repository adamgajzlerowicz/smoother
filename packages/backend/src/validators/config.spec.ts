import { validateConfig } from './config';

const successConfig = {
    "commands": [
        {
            "name": "update react",
            "command": "gitCommand",
            "fields": [],
            "payload": {
                "path": "/Users/adam/projects/react",
                "branch": "master",
                "pull": true
            }
        },
        {
            "name": "run mock command",
            "command": "anyCommand",
            "payload": {
                "command": "node /Users/adam/projects/smoother/packages/mockEmiter/index.js"
            }
        },
        {
            "name": "this will fail",
            "command": "anyCommand",
            "payload": {
                "command": "node /projects/smoother/packages/mockEmiter/index.js"
            }
        },
        {
            "name": "b -> a",
            "command": "replaceCommand",
            "payload": {
                "path": "node /projects/smoother/packages/mockEmiter/index.js",
                "replacer": "a",
                "replacee": "b"
            }
        },
        {
            "name": "a -> b",
            "command": "replaceCommand",
            "payload": {
                "path": "node /projects/smoother/packages/mockEmiter/index.js",
                "replacer": "a",
                "replacee": "b"
            }
        }

    ]
};

describe('validateConfig', () => {
    test('success', () => {
        const result = validateConfig(successConfig);
        expect(result.value).toMatchSnapshot();
        expect(result.error).toBeFalsy();
    });

    test('fails for missing commands key', () => {
        const result = validateConfig({});
        expect(result.error).toBeDefined();
    })

    test('removes extra keys', () => {
        const result = validateConfig({ foo: 'bar', commands: [] });
        expect(result.value).toEqual({ commands: [] });
    })
})
