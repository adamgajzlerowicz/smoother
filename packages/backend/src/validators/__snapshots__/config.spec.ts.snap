// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateConfig success 1`] = `
Object {
  "commands": Array [
    Object {
      "command": "gitCommand",
      "fields": Array [],
      "name": "update react",
      "payload": Object {
        "branch": "master",
        "path": "/Users/adam/projects/react",
        "pull": true,
      },
    },
    Object {
      "command": "anyCommand",
      "name": "run mock command",
      "payload": Object {
        "command": "node /Users/adam/projects/smoother/packages/mockEmiter/index.js",
      },
    },
    Object {
      "command": "anyCommand",
      "name": "this will fail",
      "payload": Object {
        "command": "node /projects/smoother/packages/mockEmiter/index.js",
      },
    },
    Object {
      "command": "replaceCommand",
      "name": "b -> a",
      "payload": Object {
        "path": "node /projects/smoother/packages/mockEmiter/index.js",
        "replacee": "b",
        "replacer": "a",
      },
    },
    Object {
      "command": "replaceCommand",
      "name": "a -> b",
      "payload": Object {
        "path": "node /projects/smoother/packages/mockEmiter/index.js",
        "replacee": "b",
        "replacer": "a",
      },
    },
  ],
}
`;
